File: server/src/main/java/com/victovito/chessapp/models/CastlesModel.java
package com.victovito.chessapp.models;

import com.victovito.chessapp.objects.Castles;

public class CastlesModel {
    AvailableCastlesModel white;
    AvailableCastlesModel black;
    
    public CastlesModel() {}

    public CastlesModel(Castles castles) {
        white = new AvailableCastlesModel(castles.getWhite());
        black = new AvailableCastlesModel(castles.getBlack());
    }

    public AvailableCastlesModel getWhite() {
        return white;
    }

    public AvailableCastlesModel getBlack() {
        return black;
    }
    
}

File: client/src/models/CastlesModel.js
import AvailableCastlesModel from "./AvailableCastlesModel";

export default class CastlesModel {
    /** @type {AvailableCastlesModel} */
    white;
    /** @type {AvailableCastlesModel} */
    black;

    /**
     * @param {CastlesModel} data 
     */
    constructor(data) {
        this.white = new AvailableCastlesModel(data.white);
        this.black = new AvailableCastlesModel(data.black);
    }
}